server/src/main/java/org/opensearch/identity/tokens/TokenManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.tokens;
+
+/**
+ * This interface defines the expected methods of a token manager
+ */
+public interface TokenManager {
+
+    /**
+     * Create a new auth token
+     * @return A new auth token
+     */
+    public AuthToken issueToken();
```

Comment:
I think when a token is generated it should be for some purpose so we should add a String parameter for the audience for the token. which is our roundabout way of saying its the name of the extension

Reply:
I will add an audience field here.